### DISCUSSION TITLE
Need help routing and initiating traffic

### DISCUSSION THREAD

**Discussion by bbe-dw** [strongswan/strongswan#1979] [Q&A]
Hi,

I'm having a flow routing problem for which I can't find the reason.
Thank you in advance for your help and ideas.

Context:
IPSec tunnel between a remote site (Fortinet firewall) and a machine hosted on Azure (Debian11 with Strongswan).

######################################
Site A = Fortinet:
Public IP: 195.130.X.Y
Lan 10.22.1.0/24

Site B = Strongswan from Azure:
Public IP: 20.19.V.Z NAT Azure to 10.41.1.4
IP Lan: 10.41.0.0/16
######################################
/etc/ipsec.conf

config setup
  cachecrls=no
  charondebug="3"
  strictcrlpolicy=no
  uniqueids=yes

conn %default
  auto=route
  type=tunnel
  authby=pubkey
  ike=aes128-sha256-modp2048,aes128-sha256-modp3072,aes128-sha256-modp4096,aes128-sha256-modp6144,aes128-sha256-modp8192,aes128-sha256-curve25519,aes128-sha384modp2048,aes128-sha384-modp3072,aes128-sha384-modp4096,aes128-sha384-modp6144,aes128-sha384-modp8192,aes128-sha384-curve25519,aes128-sha512-modp2048,aes128-sha512-modp3072,aes128-sha512-modp4096,aes128-sha512-modp6144,aes128-sha512-modp8192,aes128-sha512-curve25519,aes256-sha256-modp2048,aes256-sha256-modp3072,aes256-sha256-modp4096,aes256-sha256-modp6144,aes256-sha256-modp8192,aes256-sha256-curve25519,aes256-sha384-modp2048,aes256-sha384-modp3072,aes256-sha384-modp4096,aes256-sha384-modp6144,aes256-sha384-modp8192,aes256-sha384-curve25519,aes256-sha512-modp2048,aes256-sha512-modp3072,aes256-sha512-modp4096,aes256-sha512-modp6144,aes256-sha512-modp8192,aes256-sha512-curve25519!
  esp=aes128-sha256-modp2048,aes128-sha256-modp3072,aes128-sha256-modp4096,aes128-sha256-modp6144,aes128-sha256-modp8192,aes128-sha256-curve25519,aes128-sha384modp2048,aes128-sha384-modp3072,aes128-sha384-modp4096,aes128-sha384-modp6144,aes128-sha384-modp8192,aes128-sha384-curve25519,aes128-sha512-modp2048,aes128-sha512-modp3072,aes128-sha512-modp4096,aes128-sha512-modp6144,aes128-sha512-modp8192,aes128-sha512-curve25519,aes256-sha256-modp2048,aes256-sha256-modp3072,aes256-sha256-modp4096,aes256-sha256-modp6144,aes256-sha256-modp8192,aes256-sha256-curve25519,aes256-sha384-modp2048,aes256-sha384-modp3072,aes256-sha384-modp4096,aes256-sha384-modp6144,aes256-sha384-modp8192,aes256-sha384-curve25519,aes256-sha512-modp2048,aes256-sha512-modp3072,aes256-sha512-modp4096,aes256-sha512-modp6144,aes256-sha512-modp8192,aes256-sha512-curve25519!
  ikelifetime=1440m
  keyexchange=ikev2
  reauth=no
  lifetime=720m
  margintime=3m
  keyingtries=1
  dpdaction=restart
  dpddelay=30s
  mobike=no

# Main subnet for IPSec tunneling
conn strongswan
  auto=route
  authby=secret
  ike=aes256-sha256-modp2048!
  esp=aes256-sha256-modp2048!
  left=10.41.1.4
  leftsubnet=10.41.0.0/16
  right=195.130.X.Y
  rightsubnet=10.22.1.0/24

######################################

Status of IKE charon daemon (strongSwan 5.9.1, Linux 6.1.0-0.deb11.11-cloud-amd64, x86_64):
  uptime: 7 minutes, since Nov 06 09:50:27 2023
  malloc: sbrk 2568192, mmap 0, used 670928, free 1897264
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 3
  loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark stroke updown eap-mschapv2 xauth-generic counters
Listening IP addresses:
  10.41.1.4
Connections:
  strongswan:  10.41.1.4...195.130.X.Y  IKEv2, dpddelay=30s
  strongswan:   local:  [10.41.1.4] uses pre-shared key authentication
  strongswan:   remote: [195.130.X.Y] uses pre-shared key authentication
  strongswan:   child:  10.41.0.0/16 === 10.22.1.0/24 TUNNEL, dpdaction=restart
Routed Connections:
  strongswan{1}:  ROUTED, TUNNEL, reqid 1
  strongswan{1}:   10.41.0.0/16 === 10.22.1.0/24
Security Associations (1 up, 0 connecting):
  strongswan[1]: ESTABLISHED 7 minutes ago, 10.41.1.4[10.41.1.4]...195.130.X.Y[195.130.X.Y]
  strongswan[1]: IKEv2 SPIs: 670443fba077f77f_i af4bce6a39725643_r*, rekeying in 23 hours
  strongswan[1]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  strongswan{2}:  INSTALLED, TUNNEL, reqid 1, ESP SPIs: c0194591_i 8a8e5568_o
  strongswan{2}:  AES_CBC_256/HMAC_SHA2_256_128, 11623 bytes_i (84 pkts, 12s ago), 11356 bytes_o (69 pkts, 14s ago), rekeying in 11 hours
  strongswan{2}:   10.41.0.0/16 === 10.22.1.0/24

**No IPtables filtering in place.**
######################################

Problem:

If I ping a machine from site A (10.22.1.21) to site B (10.41.1.4)
Result: Ping NOK. :(

If I ping a machine from site B (10.41.1.4) to site A (10.22.1.21)
Result: Ping OK. :)

If I ping again from machine 10.22.1.21 to 10.41.1.4 it now works! :/
As long as there's traffic, it's routed correctly. As soon as there is no more traffic, after a few minutes it is again impossible to reach 10.41.1.4 from 10.22.1.0/24.

Final situation:
Phase 1 and 2 are well established.
If traffic is initiated from 10.41.1.4 to 10.22.1.0/24, it works.
If traffic is initiated from 10.22.1.0/24 to 10.41.1.4 it does not work. Traffic must first pass in the other direction.

Thanks for your ideas
I've already thought of pinging 10.41.1.4 to 10.22.1.1, but I'd like to find out the cause of my problem.

Benjamin

**Comment by tobiasbrunner** [comment]
You should check the traffic counters on both ends after establishing the SA and sending traffic only from A to B. And then send from B to A and check again.
